torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3464,6 +3507,8 @@ def _finalize_params(fsdp_module: FullyShardedDataParallel) -> None:
             delattr(self, "_handles_exec_order")
             self._register_param_handles_from_handles(handles_per_flat_param)
             self._param_exec_order_state = ParamExecOrderState.INITIALIZED
+            # Force it to reset
```

Comment:
we need to `lazy_init` again so that parameters have needed attributes, e.g. `_post_backward_called `